### ISSUE TITLE
Install `@tanstack/react-query` and build error

### ISSUE THREAD

**Issue by callqh** [farm-fe/farm#1275]
### Steps to reproduce

[Reproduction repository ](https://github.com/jstors/farm-react-admin/tree/feature/mock)

run `pnpm build`
### What is expected?

build success

### What is actually happening?

```shell
thread '<unnamed>' panicked at crates/plugin_runtime/src/render_resource_pot/source_replacer.rs:138:11:
Cannot find module id for source "./useBaseQuery.js" from ModuleId { relative_path: "node_modules/.pnpm/@tanstack+react-query@5.32.0_react@18.2.0/node_modules/@tanstack/react-query/build/modern/useInfiniteQuery.js", query_string: "" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### System Info

```Shell
  System:
    OS: macOS 14.4.1
    CPU: (10) arm64 Apple M2 Pro
    Memory: 193.11 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.20.1 - ~/.nvm/versions/node/v16.20.1/bin/node
    Yarn: 1.22.22 - ~/Library/pnpm/yarn
    npm: 8.19.4 - ~/.nvm/versions/node/v16.20.1/bin/npm
    pnpm: 8.15.5 - ~/Library/pnpm/pnpm
    bun: 0.6.13 - ~/.bun/bin/bun
  Browsers:
    Chrome: 122.0.6261.112
    Edge: 124.0.2478.51
    Safari: 17.4.1
  npmPackages:
    @farmfe/core: ^1.1.2 => 1.1.2 
```

**Comment by ante-kalajzic-deel** [comment]
Just want to add, not isolated to react-query, happens with some other dependencies too (in my case Braze).

**Comment by wre232114** [comment]
It will be fixed in #1284
